### PR TITLE
Add flag MICROPY_HW_FLASH_DQS

### PR DESCRIPTION
**Pull request by d3v1c3nv11**
Add flag MICROPY_HW_FLASH_DQS in mpconfigboard.mk, values as defined in FlexSPI Read Sample Clock Source definition 
example:
MICROPY_HW_FLASH_DQS = kFlexSPIReadSampleClk_LoopbackInternally

### CODE REVIEW

**Comment by robert-hh** [comment]
Thanks. That might be the change for now. But I had the impression that you asked to a more generic change. 
Note: The code formatting does not match the requestes style. Better use :
```
        #if defined(MICROPY_HW_FLASH_DQS)
        .readSampleClkSrc = MICROPY_HW_FLASH_DQS,
        #else
        .readSampleClkSrc = kFlexSPIReadSampleClk_LoopbackFromDqsPad,
        #endif
```
Spaces instead of tabs, indent the preprocessor statements.
        #endif

File: ports/mimxrt/hal/qspi_nor_flash_config.c [hunk 1]
```diff
@@ -29,7 +29,11 @@ const flexspi_nor_config_t qspiflash_config = {
     {
         .tag = FLEXSPI_CFG_BLK_TAG,
         .version = FLEXSPI_CFG_BLK_VERSION,
+#if defined(MICROPY_HW_FLASH_DQS)
+		.readSampleClkSrc = MICROPY_HW_FLASH_DQS,
```
Comment (dpgeorge) [inline ports/mimxrt/hal/qspi_nor_flash_config.c:33]: IMO it's cleaner to have at the top of this file something like:
```c
#ifndef MICROPY_HW_FLASH_DQS
#define MICROPY_HW_FLASH_DQS kFlexSPIReadSampleClk_LoopbackFromDqsPad
#endif
```
Then the code here is simply:
```c
.readSampleClkSrc = MICROPY_HW_FLASH_DQS,
```

Comment (d3v1c3nv11) [reply ports/mimxrt/hal/qspi_nor_flash_config.c:33]: i agree this way is cleaner

**Comment by robert-hh** [comment]
@d3v1c3nv11 Will you update your PR?

**Comment by robert-hh** [comment]
I do not know if you did that already, but I sketched a set of board files for the RT1010Py. Using these I can compile a firmware image. So at least the pins assignments in mpconfigboard.h do not create a compile error. I tried to follow the labeling of the sample board. There is one potential source for misunderstanding: The ID numbers for I2C and SPI in Python are 0 and 1. At the board they are labeled 1 and 2.

[OLIMEX_RT1010Py.zip](https://github.com/micropython/micropython/files/7730122/OLIMEX_RT1010Py.zip)

**Comment by dpgeorge** [comment]
Looks like there's a new PR #8093.

**Comment by robert-hh** [comment]
I noticed that yesterday. So this one could be closed.

**Comment by d3v1c3nv11** [comment]
> I do not know if you did that already, but I sketched a set of board files for the RT1010Py. Using these I can compile a firmware image. So at least the pins assignments in mpconfigboard.h do not create a compile error. I tried to follow the labeling of the sample board. ~There is one potential source for misunderstanding: The ID numbers for I2C and SPI in Python are 0 and 1. At the board they are labeled 1 and 2.~
> 
> [OLIMEX_RT1010Py.zip](https://github.com/micropython/micropython/files/7730402/OLIMEX_RT1010Py.zip)

look at mine fork https://github.com/d3v1c3nv11/micropython.git there i added the board, but have problem with second REPL

**Comment by robert-hh** [comment]
Looks similar. This morning I realize, that temporarily you could replicate the first entry for SPI and I2C, into:
`#define MICROPY_HW_SPI_INDEX { 2, 2, 1 }`
and
`#define MICROPY_HW_I2C_INDEX   { 1, 1, 2 }`
That avoids a memory error, in case someone uses ID 0 for I2C and SPI. SPI0 and SPI1 would point to the same device then. A fix for the 0 index is prepared and will come with the next 'service' PR. You can see that already here: https://github.com/robert-hh/micropython/commit/78424499c8c2189fcbb514e2ef22040b4531361c
With that PR you can keep 0 as fist entry in the 
Besides that, your board files look pretty similar to the one I had made. Not surprising since they are based on the same schematics.

**Comment by robert-hh** [comment]
I have two questions about your mpconfigboard.mk file:
a) You include hal/flexspi_nor_flash.c to SRC_C. But that is already done in Makefile.
b) You have set the flash size to 2 MB and a reserved size of 4K, similar to Teensy 4.x. I assume it's intentional to support a kind of firmware loader. 

b.t.w.: We are working on a generic bootloader for the mimxrt devices.

**Comment by d3v1c3nv11** [comment]
> I have two questions about your mpconfigboard.mk file: a) You include hal/flexspi_nor_flash.c to SRC_C. But that is already done in Makefile. b) You have set the flash size to 2 MB and a reserved size of 4K, similar to Teensy 4.x. I assume it's intentional to support a kind of firmware loader.
> 
> b.t.w.: We are working on a generic bootloader for the mimxrt devices.

this are not the only copy-paste bugs to clean :) please contact us at support@olimex.com

**Comment by dpgeorge** [comment]
Superseded by #8093.

**Pull request closed without merging**